.github/workflows/cypress-Run-all-tests-no-cache.yml

```diff
@@ -69,32 +76,49 @@ jobs:
       - name: Use Node.js
         uses: actions/setup-node@v2
         with:
-          node-version: 14
+          node-version: 16
 
+      # On push or when we specify we don't want to send the tests to Dashboard through workflow dispatch
+      # we set the environment variables dynamically to empty in order to avoid
+      # recording the test execution to Cypress Dashboard.
+      # - name: Set Dashboard flags to empty when push or worflow_dispatch
+      #  if: ${{ github.event_name == 'push' || github.event.inputs.cyDashboard == 'No' }}
+      #  run: |
+      #    echo "CY_RECORD_KEY=" >> $GITHUB_ENV
+      #    echo "CY_RECORD_FLAG=" >> $GITHUB_ENV
+
+      # We define the environment variables to send the test execution to the Cypress Dashboard
+      # on every pull request or when specifying to record the tests on workflow dispatch
+      # - name: Set Dashboard flags when pull_request or worflow_dispatch
+      #  if: ${{ github.event_name == 'pull_request' || github.event.inputs.cyDashboard == 'Yes' }}
+      #  run: |
+      #    echo "CY_RECORD_KEY=${{ secrets.CYPRESS_RECORD_KEY }}" >> $GITHUB_ENV
+      #    echo "CY_RECORD_FLAG=-- --record --key " >> $GITHUB_ENV
+     
       - name: Run tests by browser
         uses: cypress-io/github-action@v2
         timeout-minutes: 10
         with:
           # 'build' starts the defauld demo 
           build: npm run build
-          # 'test:ci' runs tests over Docker image for build context 
-          command: npm run test:ci
-          record: false
-          parallel: false
+          # 'test:ci' runs tests over Docker image for build context
+          # we also send the Cypress Dashboard record key dynamically
+          command: npm run test:ci # ${{ env.CY_RECORD_FLAG }} ${{ env.CY_RECORD_KEY }}
+          record: true
+          parallel: true
           group: "${{ matrix.type }} tests on ${{ matrix.browser }}"
           browser: ${{ matrix.browser }}
           config: "video: true"
           spec: |
             cypress/tests/**/*.js
           # config: "\"video\": true, \"screenshotsFolder\": \"cypress/screenshots/${{ matrix.browser }}\", \"videosFolder\": \"cypress/videos/${{ matrix.browser }}\", \"video\": true"
           # spec: cypress/tests/${{ matrix.type }}/*.js
-        # env:
+        env:
         # https://github.com/cypress-io/cypress-example-kitchensink/settings/secrets
-            # screenshotsFolder: "cypress/screenshots/${{ matrix.browser }}"
-            # videosFolder: "cypress/videos/${{ matrix.browser }}"
+        #     # screenshotsFolder: "cypress/screenshots/${{ matrix.browser }}"
+        #     # videosFolder: "cypress/videos/${{ matrix.browser }}"
```

Comment:
Watch out for things like this in PRs: if the behavior hasn't changed, try not to commit unnecessary changes (like the double comments).

Reply:
> This Pull Request includes the necessary documentation for this project on the testing integration with Cypress Dashboard, such as:
> 
>     * When should the tests be sent to the Cypress Dashboard.
This Pull Request includes the necessary documentation for this project on the testing integration with Cypress Dashboard, such as:

>    When should the tests be sent to the Cypress Dashboard.
>    How to send local executions to run on Cypress Dashboard through GitHub actions.

These goals above are out of the scope of this task and therefore they shouldn't be considered on this pull request.